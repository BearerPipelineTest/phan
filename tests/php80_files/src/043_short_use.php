<?php
class A952 {
    public function main(): Closure {
        return static fn () => $this;
    }
}
(new A952())->main()();
