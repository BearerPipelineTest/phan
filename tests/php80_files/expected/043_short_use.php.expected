%s:4 PhanCompatibleArrowFunction Cannot use arrow functions before php 7.4 in (fn)
%s:4 PhanPluginUnknownClosureReturnType Closure Closure() has no declared or inferred return type (Types inferred after analysis: object)
%s:4 PhanUndeclaredThis Variable $this is undeclared (Did you mean (function instead of static function for Closure() at line 4))
%s:4 PhanUnreferencedClosure Possibly zero references to Closure()
